internal/config/config.go

```diff
@@ -56,10 +56,10 @@ func Load(config *api.Config) error {
 			return err
 		}
 	}
-	if err := config.Source.Repo.SetBasicAuth(os.Getenv("SOURCE_AUTH_USERNAME"), os.Getenv("SOURCE_AUTH_PASSWORD")); err != nil {
+	if err := config.GetSource().GetRepo().SetBasicAuth(os.Getenv("SOURCE_AUTH_USERNAME"), os.Getenv("SOURCE_AUTH_PASSWORD")); err != nil {
```

Comment:
you are not checking if the source has a `repo` in this method

Reply:
right! however it is not failing 🤔 maybe the getter is managing the case of repo being nil?